compass/_scrapers/hierarchy.py

```diff
@@ -0,0 +1,98 @@
+import datetime
+import json
+
+import requests
+from lxml import html
+
+from compass.interface_base import CompassInterfaceBase
+from compass.settings import Settings
+from compass.utility import compass_restify
+
+endpoints = {
+    "Countries": "/countries",
+    "HQ Sections": "/hq/sections",
+    "Regions": "/regions",
+    "Country Sections": "/country/sections",
+    "Counties": "/counties",
+    "Region Sections": "/region/sections",
+    "Districts": "/districts",
+    "County Sections": "/county/sections",
+    "Groups": "/groups",
+    "District Sections": "/district/sections",
+    "Group Sections": "/group/sections",
+}
+
+
+class CompassHierarchyScraper(CompassInterfaceBase):
+    def __init__(self, session: requests.Session):
+        super().__init__(session)
+
+    # see CompassClient::retrieveLevel or retrieveSections in PGS\Needle php
+    def get_units_from_hierarchy(self, parent_unit: int, level: str) -> list:
+        """Get all children of a given unit
+
+        TODO can we do this without needing to provide the level string?
+
+        """
+
+        # Get API endpoint from level
+        level_endpoint = endpoints[level]
+
+        # TODO PGS\Needle has `extra` bool in func signature to turn LiveData on/off
+        result = self._post(f"{Settings.base_url}/hierarchy{level_endpoint}", json={"LiveData": "Y", "ParentID": f"{parent_unit}"})
+        result_json = result.json()
+
+        # Handle unauthorised access
+        if result_json == {"Message": "Authorization has been denied for this request."}:
+            return [{"id": None, "name": None}]
+
+        result_units = []
+        for unit_dict in result_json:
+            # TODO LiveData check here too
+            parsed = {
+                "id": int(unit_dict["Value"]),
+                "name": unit_dict["Description"],
+                "parent_id": unit_dict["Parent"],
+            }
+            if unit_dict["Tag"]:
+                tag = json.loads(unit_dict["Tag"])
+                parsed["status"] = tag[0]["org_status"]
+                parsed["address"] = tag[0]["address"]
+                parsed["member_count"] = tag[0]["Members"]
+                if "SectionTypeDesc" in tag[0]:
+                    parsed["section_type"] = tag[0]["SectionTypeDesc"]
```

Comment:
Consider using .get() to access the key on the tag obj, unless we're sure these entries are always going to be set. Would expect there to be some entries missing addresses for example.